### PR TITLE
65 add tos

### PR DESCRIPTION
**Pull request by AlSchlo**
Add the terms of services according to #65 
Refactored presentation code & adapted the test to increase the coverage.
Nothing particular to note, quite straighforward code (references to StackOverFlow posts have been added)

Note: Some ugly code for the ToS launch, as Espresso cannot click on text within a checkBox. This had to be done by manually entering the coordinates of the text within the checkBox.

### CODE REVIEW

**Review by emduc — APPROVED** [review]
Some nice features thank you. Neat presentation and code, some few naming / comments details here and there but the global structure is nice.

File: app/src/main/java/com/github/h3lp3rs/h3lp/SignInActivity.kt [hunk 1]
```diff
@@ -19,6 +24,8 @@ import com.google.firebase.auth.GoogleAuthProvider
 import com.google.firebase.auth.ktx.auth
 import com.google.firebase.ktx.Firebase
 
+// TODO: We should reach a consensus on where to put these things
+const val ORIGIN: String = "ORIGIN"
```
[the remark this answers is not in the document]
Comment (AlSchlo) [reply app/src/main/java/com/github/h3lp3rs/h3lp/SignInActivity.kt:28]: Mhh, multiple activities use/receive this!

File: app/src/main/res/layout/activity_presentation_irrelevant.xml [hunk 2]
```diff
@@ -148,4 +148,17 @@
         app:srcCompat="@drawable/red_cross"
         android:contentDescription="@string/red_cross" />
 
+    <CheckBox
+        android:id="@+id/pres3_checkBox"
+        android:layout_width="138dp"
+        android:layout_height="44dp"
+        android:contentDescription="Accept_TOS"
+        android:text="Accept our ToS"
```
[the remark this answers is not in the document]
Comment (AlSchlo) [reply app/src/main/res/layout/activity_presentation_irrelevant.xml:156]: Yes, will change that (and other CodeClimate warnings (didn't have the time yet).

File: app/src/androidTest/java/com/github/h3lp3rs/h3lp/presentation/PresIrrelevantActivityTest.kt [hunk 3]
```diff
@@ -3,50 +3,201 @@ package com.github.h3lp3rs.h3lp.presentation
...
+    // Since this is not a textView, espresso cannot click on text in it. So we manually find the
+    // right coordinates of the highlighted text.
+    private fun clickPercent(pctX: Float, pctY: Float): ViewAction {
+        return GeneralClickAction(
+            Tap.SINGLE,
+            { view ->
+                val screenPos = IntArray(2)
+                view.getLocationOnScreen(screenPos)
+                val w = view.width
+                val h = view.height
```
Comment (Adrinels) [inline app/src/androidTest/java/com/github/h3lp3rs/h3lp/presentation/PresIrrelevantActivityTest.kt:56]: Don't Know if here h and v are really useful

File: app/src/main/java/com/github/h3lp3rs/h3lp/preferences/Preferences.kt [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package com.github.h3lp3rs.h3lp.preferences
+
+import android.content.Context
+import androidx.appcompat.app.AppCompatActivity
+
+/**
+ * Convenience class to store the global app preferences
+ */
+class Preferences(private val file: Files, context: Context) {
+
```
Comment (Adrinels) [inline app/src/main/java/com/github/h3lp3rs/h3lp/preferences/Preferences.kt:10]: Nice Idea, we will have to add more function for more type later on.

**Review by Adrinels — APPROVED** [review]
Nice! Nothing to add to Emilien comment.
Good code and really good modularization

File: app/src/androidTest/java/com/github/h3lp3rs/h3lp/presentation/PresIrrelevantActivityTest.kt (hunk 3 above)
Comment (AlSchlo) [reply app/src/androidTest/java/com/github/h3lp3rs/h3lp/presentation/PresIrrelevantActivityTest.kt:56]: Indeed, CC'd the SOF code.
Will refactor at next sprint.

**Pull request merged**